Assets/Scripts/Movement/CameraLook.cs

```diff
@@ -26,22 +30,88 @@ void Start()
     }
     private void OnEnable()
     {
-        LookAtJarvis.AddListener(FixLook);
-        LookFree.AddListener(FreeLook);
+        lookAtEvent.AddListener(LookAt);
     }
 
-    void FixLook()
+    private void LookAt(Vector3 targetPos)
     {
-        _fixlook = true;
+        _lookAtRoutine = LookAtRoutine(targetPos);
+        StartCoroutine(_lookAtRoutine);
     }
-    void FreeLook()
+
+    private void GetShortestRotationTarget(float source, float target, out float bestSource, out float bestTarget)
     {
+        source = mod(source, 360);
+        target = mod(target, 360);
+
+        float invertedSource = source - 360;
+        float invertedTarget = target - 360;
+        // figure out if we should invert target or source
+        if (target < source)
+        {
+            bestTarget = target;
+            if (source - target < target - invertedSource)
+            {
+                bestSource = source;
+                return;
+            }
+            else
+            {
+                bestSource = invertedSource;
+                return;
+            }
+        }
+        else
+        {
+            bestSource = source;
+            if (target - source < source - invertedTarget)
+            {
+                bestTarget = target;
+                return;
+            }
+            else
+            {
+                bestTarget = invertedTarget;
+                return;
+            }
+        }
+
+
+    }
+
+    private IEnumerator LookAtRoutine(Vector3 targetPos)
+    {
+        _fixlook = true;
+        float t = 0.0f;
+        
+        Vector3 targetRotation = Quaternion.LookRotation(targetPos - this.transform.position).eulerAngles;
+        Vector3 sourceRotation = new Vector3(
+            mod(transform.localRotation.eulerAngles.x,360), 
+            mod(player.transform.rotation.eulerAngles.y, 360), 
+            0);
+
+        float sourceXRotation, targetXRotation;
+        float sourceYRotation, targetYRotation;
+        GetShortestRotationTarget(sourceRotation.x, targetRotation.x, out sourceXRotation, out targetXRotation);
+        GetShortestRotationTarget(sourceRotation.y, targetRotation.y, out sourceYRotation, out targetYRotation);
+        Debug.Log(new Vector3(sourceXRotation, sourceYRotation, 0));
+        Debug.Log(new Vector3(targetXRotation, targetYRotation, 0));
```

Comment:
```suggestion
```
This will probably fill up the debug log since it triggers whenever the player looks at something I am assuming.